beluga_amcl/include/beluga_amcl/filter_update_control/filter_update_control_mixin.hpp

```diff
@@ -0,0 +1,192 @@
+    const typename SelectiveResamplingPolicy::param_type & selective_resampling_config,
+    Rest &&... rest)
+  : Mixin(std::forward<Rest>(rest)...),
+    update_when_moving_policy_{resample_on_motion_config},
+    resample_interval_policy_{resample_interval_config},
+    selective_resampling_policy_{selective_resampling_config} {}
+
+  /** @brief Execute a filter update step, conditioned by the installed resampling policies.
+   *  @param odom_to_base_transform Odom-to-base transform, for the motion update.
+   *  @param laser_scan Laser-scan information, for the sensor update.
+   *  @return true if the estimation result of the filter has been updated, false otherwise.
+   */
+  [[nodiscard]] bool update_filter(
+    const motion_update_type & odom_to_base_transform,
+    sensor_update_type && laser_scan) final
+  {
+    return this->update_filter_impl(
+      std::execution::sequenced_policy{}, odom_to_base_transform,
+      std::move(laser_scan));
+  }
+
+  /**
+   * \overload
+   * Overload for performing a full filter update using the sequential execution policy.
+   */
+  [[nodiscard]] bool update_filter(
+    std::execution::sequenced_policy exec_policy,
+    const motion_update_type & odom_to_base_transform,
+    sensor_update_type && laser_scan) final
+  {
+    return this->update_filter_impl(
+      exec_policy, odom_to_base_transform,
+      std::move(laser_scan));
+  }
+
+  /**
+   * \overload
+   * Overload for performing a full filter update using the parallel execution policy.
+   */
+  [[nodiscard]] bool update_filter(
+    std::execution::parallel_policy exec_policy,
+    const motion_update_type & odom_to_base_transform,
+    sensor_update_type && laser_scan) final
+  {
+    return this->update_filter_impl(
+      exec_policy, odom_to_base_transform,
+      std::move(laser_scan));
+  }
+
+private:
+  UpdateFilterWhenMovingPolicy update_when_moving_policy_;
+  ResampleIntervalPolicy resample_interval_policy_;
+  SelectiveResamplingPolicy selective_resampling_policy_;
+
+  template<typename ExecutionPolicy>
+  [[nodiscard]] bool update_filter_impl(
+    ExecutionPolicy exec_policy,
+    const motion_update_type & odom_to_base_transform,
+    sensor_update_type && laser_scan)
+  {
+    bool filter_updated = false;
+
+    // based on NAV2 AMCL's decision tree to determine when to update, resample
+    // and whether to assume that the estimation has been updated.
+    const auto far_enough_to_update = update_when_moving_policy_.do_filter_update(
+      odom_to_base_transform);
+    if (far_enough_to_update) {
+      this->self().update_motion(odom_to_base_transform);
+      this->self().sample(exec_policy);
+
+      this->self().update_sensor(std::move(laser_scan));
+      this->self().reweight(exec_policy);
+
+      const auto resample_rate_divider_agrees =
+        resample_interval_policy_.do_resampling();
+
+      if (resample_rate_divider_agrees) {
+        filter_updated = true;
```

Comment:
I know there is a comment above explaining that this is based on Nav2 AMCL's decision tree. Maybe we can add another comment here to clarify that this is not a bug and the filter might have been updated and still return false.

Reply:
Done. Also changed the variable name (and the return value documentation) that this is meant to indicate whether the filter estimate can be expected to have changed in significant way.